### ISSUE TITLE
R 4.2: Support new pipe placeholder syntax

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#10757]
Coming to R 4.2 is the new pipe with now a placeholder syntax 

New pipe support has been added in 
* https://github.com/rstudio/rstudio/pull/8533
* https://github.com/rstudio/rstudio/pull/8744

Now you can also use a paceholder 
````r
mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)
````
Example here: https://twitter.com/henrikbengtsson/status/1501306369319735300
Added in : https://github.com/wch/r-source/commit/587244ff8725aa31ebc23ede6ea3be5575417209

Currently, the IDE is throwing a "Unexpected token error"
![image](https://user-images.githubusercontent.com/6791940/157430597-e302cc06-c1bc-44db-a05a-477e2b8bc36b.png)

Opening the issue just to be sure it is on your radar.

**Comment by ronblum** [comment]
Triage: Would this need to be backported?

**Comment by jmcphers** [comment]
Yes, we'd want to backport it to supported releases.

**Comment by kevinushey** [comment]
I'll take a look.

**Comment by mikebessuille** [comment]
we want to backport this (PT includes some support for R4.2) but don't need to urgently backport this into 2022.02.1

**Comment by ronblum** [comment]
Testing with the following examples (the results are a bit different, details to follow):

*R Source*
````
# Example of of native pipe placeholder orig. from pipeOp
mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)

# Example from QA notes
mtcars |> summary(object = _)

# Making sure that putting the entire statement in a function doesn't matter
fc <- function(c){
  mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)
}
fc(1)

# Also making sure that putting the statement in a function doesn't matter
fd <- function(d){
  mtcars |> summary(object = _)
}
fc(1)

# Using a user-defined function
fab <- function(a, b){
  print(b)
}
12 |> fab(1, b = _)

````

*R Markdown*
````
---
title: "Placeholder"
author: "New Pipe"
date: '2022-04-14'
output: html_document
---

```{r}
# Example of of native pipe placeholder orig. from pipeOp
mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)

# Example from QA notes
mtcars |> summary(object = _)

# Making sure that putting the entire statement in a function doesn't matter
fc <- function(c){
  mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)
}
fc(1)

# Also making sure that putting the statement in a function doesn't matter
fd <- function(d){
  mtcars |> summary(object = _)
}
fc(1)

# Using a user-defined function
fab <- function(a, b){
  print(b)
}
12 |> fab(1, b = _)
```
````

**Comment by ronblum** [comment]
Tested in:
- RStudio Desktop 2022.02.2+481 & 2022.6.0+296 on MacOS 12.3 with R 4.2.0
- RStudio Server 2022.6.0+296 on Ubuntu 20.04 with R 4.1.2

[Ghost Orchid patch 3 TBD]

Results:
1) The new native pipe operator's placeholder (`_`) is no longer identified as an ![image](https://user-images.githubusercontent.com/31009952/163464162-ea715b57-3575-4bdd-86f7-08bab2815a1e.png) error 
 (cf. to 2022.02.1+461 as a baseline).
2) There's no difference between the R script and R Markdown.
3) However, while this example is treated as fine:
`mtcars |> subset(cyl == 4) |> lm(mpg ~ disp, data = _)`
this example has a "no symbol named '_' in scope" ![image](https://user-images.githubusercontent.com/31009952/163464402-2db57444-bc8c-405a-95c9-0be072d6fdca.png) warning associated with it:
`mtcars |> summary(object = _)`

<img width="621" alt="placeholder_warning" src="https://user-images.githubusercontent.com/31009952/163463717-e8394661-a56c-4c85-9745-00cfec592aa8.png">

**Comment by ronblum** [comment]
I _think_ we don't want to show this warning, either, so putting this back into "in progress." But I can open a separate issue for this instead, since at least we're no longer flagging this as an error.

**Comment by ronblum** [comment]
Verified fixed in RStudio Desktop 2021.09.3-395, 2022.02.2-485, and 2022.06.0-313.
